### ISSUE TITLE
Can LlamaGen predict a [EOS] token when inferencing?

### ISSUE THREAD

**Issue by Xiaoyuan-Isaac-Wang** [FoundationVision/LlamaGen#44]

**Comment by daiyixiang666** [comment]
No， we donot need to

**Comment by Xiaoyuan-Isaac-Wang** [comment]
If I simply add a [eos] token after tokenizing a image for training, and when inferencing, if [eos] is predicted, the model then stops generating. Will it work? What's your thought on this, thx!

**Comment by daiyixiang666** [comment]
emm.. It will work but it will be just useless since the model will do the exact step during inference since the image has a fix latent size.

**Comment by daiyixiang666** [comment]
But it will be usefull if you want to training on different spatial ratio image and add the information as the start token

**Comment by PeizeSun** [comment]
Hi~
Adding a special token to enable different aspect ratio image generation is very promising. We will try this idea if possible in the future. Thx !

**Comment by daiyixiang666** [comment]
Yes, And my another suggest based on my training in many t2i models, add cross attention instead of add the token to the front will produce more promising result.
